Error when compiling

When cross-compiling for the EdgeRouter Lite using the Cavium SDK (3.1), I get the following error.

```
lochnair@debianbuild:~/edgeos/sch_cake$ ARCH=mips CROSS_COMPILE=mips64-octeon-linux-gnu- make
make -C ../linux-src/linux-3.10/kernel/ SUBDIRS=/home/lochnair/edgeos/sch_cake modules LDFLAGS_MODULE="--build-id=0x417e6b97179c02996dabfccd87e9317f21037d6d" CFLAGS_MODULE="-DCAKE_VERSION=\\\"417e6b97179c02996dabfccd87e9317f21037d6d\\\""
make[1]: Entering directory `/home/lochnair/edgeos/linux-src/linux-3.10/kernel'
  CC [M]  /home/lochnair/edgeos/sch_cake/sch_cake.o
In file included from /home/lochnair/edgeos/sch_cake/cobalt.c:50:0,
                 from /home/lochnair/edgeos/sch_cake/sch_cake.c:59:
/home/lochnair/edgeos/sch_cake/cobalt.h: In function 'cobalt_get_time':
/home/lochnair/edgeos/sch_cake/cobalt.h:64:2: error: implicit declaration of function 'ktime_get_ns' [-Werror=implicit-function-declaration]
In file included from /home/lochnair/edgeos/sch_cake/cobalt.c:60:0,
                 from /home/lochnair/edgeos/sch_cake/sch_cake.c:59:
/home/lochnair/edgeos/sch_cake/codel5_compat.h: At top level:
/home/lochnair/edgeos/sch_cake/codel5_compat.h:53:0: warning: "codel_stats_copy_queue" redefined [enabled by default]
In file included from /home/lochnair/edgeos/sch_cake/cobalt.c:50:0,
                 from /home/lochnair/edgeos/sch_cake/sch_cake.c:59:
/home/lochnair/edgeos/sch_cake/cobalt.h:58:0: note: this is the location of the previous definition
In file included from /home/lochnair/edgeos/sch_cake/cobalt.c:60:0,
                 from /home/lochnair/edgeos/sch_cake/sch_cake.c:59:
/home/lochnair/edgeos/sch_cake/codel5_compat.h:54:0: warning: "codel_watchdog_schedule_ns" redefined [enabled by default]
In file included from /home/lochnair/edgeos/sch_cake/cobalt.c:50:0,
                 from /home/lochnair/edgeos/sch_cake/sch_cake.c:59:
/home/lochnair/edgeos/sch_cake/cobalt.h:59:0: note: this is the location of the previous definition
cc1: some warnings being treated as errors
make[2]: *** [/home/lochnair/edgeos/sch_cake/sch_cake.o] Error 1
make[1]: *** [_module_/home/lochnair/edgeos/sch_cake] Error 2
make[1]: Leaving directory `/home/lochnair/edgeos/linux-src/linux-3.10/kernel'
make: *** [default] Error 2
```

It complains about this code. As I understand this function (ktime_get_ns) is not available on this kernel version.

``` c
static inline cobalt_time_t cobalt_get_time(void)
{
    return ktime_get_ns();
}
```

Looking further I found that codel5_compat.h defines this function for older kernels, but it doesn't work:

``` c
#if KERNEL_VERSION(3, 17, 0) > LINUX_VERSION_CODE

#define ktime_get_ns() ktime_to_ns(ktime_get())

#endif
```

So for the sake of testing I changed the cobalt_get_time function to use the old method:

``` c
static inline cobalt_time_t cobalt_get_time(void)
{
    return ktime_to_ns(ktime_get());
}
```

Now it compiles with no errors. As far as I can understand with my limited C knowledge, the order the headers are included is wrong, but I have no idea how to fix that.

--- a/cobalt.c
+++ b/cobalt.c
@@ -56,10 +56,6 @@
  * more effective on unresponsive flows.
  */
 
-#if KERNEL_VERSION(3, 18, 0) > LINUX_VERSION_CODE
-#include "codel5_compat.h"
-#endif
-
 struct cobalt_skb_cb *get_cobalt_cb(const struct sk_buff *skb)
 {
 	qdisc_cb_private_validate(skb, sizeof(struct cobalt_skb_cb));

--- a/cobalt.h
+++ b/cobalt.h
@@ -58,6 +58,10 @@ typedef s64 cobalt_tdiff_t;
 #define codel_stats_copy_queue(a, b, c, d) gnet_stats_copy_queue(a, b, c, d)
 #define codel_watchdog_schedule_ns(a, b, c) qdisc_watchdog_schedule_ns(a, b, c)
 
+#if KERNEL_VERSION(3, 18, 0) > LINUX_VERSION_CODE
+#include "codel5_compat.h"
+#endif
+
 static inline cobalt_time_t cobalt_get_time(void)
 {
 	return ktime_get_ns();

